static/js/stream_edit.js

```diff
@@ -557,12 +522,25 @@ export function initialize() {
 
         $(".changeable-settings").replaceWith(change_general_settings);
         change_stream_message_retention_days_block_display_property(value);
+
+        $(".realm_default").on("change", () => {
+            $(".stream-message-retention-days-input").hide();
+        });
+        $(".unlimited").on("change", () => {
+            $(".stream-message-retention-days-input").hide();
+        });
+        $(".retain_for_period").on("change", () => {
+            $(".stream-message-retention-days-input").show();
```

Comment:
Instead of having this three differen "change" handlers, can we add a same class to all those inputs, and then write a single "change" handler which calls `change_stream_message_retention_days_block_display_property`.